### PR TITLE
le_tc/ched: Add corner cases for sched TC's

### PR DESCRIPTION
**Pull request by thapav**
This patch adds corner cases for following API's:
waitid()
waitpid()
sched_setparam()
sched_getparam()
sched_setpriority()
sched_setscheduler()
sched_rr_get_interval()

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/tc_sched.c [hunk 1]
```diff
@@ -135,6 +159,23 @@ static void tc_sched_sched_setget_scheduler_param(void)
...
+	/*  Check null/invalid priority parameter */
+
+	ret_chk = sched_setparam(getpid(), NULL);
+	TC_ASSERT_EQ("sched_setparam", ret_chk, ERROR);
+
+	ret_chk = sched_getparam(getpid(), NULL);
+	TC_ASSERT_EQ("sched_getparam", ret_chk, ERROR);
+
+	ret_chk = sched_setscheduler(0, SCHED_RR, &st_setparam);
+	TC_ASSERT_EQ("sched_setscheduler", ret_chk, ERROR);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_sched.c]: sched_setscheduler will return ERROR when priority is not in valid range. So I think you'd better set st_setparam explicitly for readability.

File: apps/examples/testcase/le_tc/kernel/tc_sched.c [hunk 2]
```diff
@@ -274,15 +330,34 @@ static void tc_sched_waitid(void)
...
 	child_pid = task_create("tc_waitid", SCHED_PRIORITY_DEFAULT, TASK_STACKSIZE, function_waitid, (char * const *)NULL);
 	TC_ASSERT_GT("task_create", child_pid, 0);
 
 	ret_chk = waitid(P_PID, child_pid, &info, WEXITED);
 	TC_ASSERT_NEQ("waitid", ret_chk, ERROR);
 	TC_ASSERT_EQ("waitid", info.si_pid, child_pid);
 
+	ret_chk = waitid(P_PID, child_pid, &info, 0);
+	TC_ASSERT_EQ("waitid", ret_chk, ERROR);
+	TC_ASSERT_EQ("waitid", errno, ENOSYS);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_sched.c]: This case is only when CONFIG_DEBUG is enabled. isn't it?

File: apps/examples/testcase/le_tc/kernel/tc_sched.c [hunk 3]
```diff
@@ -524,6 +618,32 @@ static void tc_sched_sched_getstreams(void)
...
+* @return               :void
+*/
+
+static void tc_sched_sched_setpriority(void)
+{
+	int ret_chk = ERROR;
+	FAR struct tcb_s tcb;
+
+	ret_chk = sched_setpriority(&tcb, SCHED_PRIORITY);
+	TC_ASSERT_EQ("sched_setpriority", ret_chk, OK);
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_sched.c]: you didn't set tcb variable, so I don't think this is a good testcase or not.

**Comment by sunghan-chang** [comment]
@thapav Could you fill details on commit message like calling some function hundred times, put invalid arg , etc?

**Comment by thapav** [comment]
@sunghan-chang @jeongchanKim 
Modified as per review comments, Please check

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**